include/picongpu/fields/MaxwellSolver/DirSplitting/DirSplitting.hpp

```diff
@@ -19,170 +19,178 @@
-        dc.releaseData( FieldB::getName() );
-    }
-
-    void update_afterCurrent(uint32_t) const
-    {
-        DataConnector &dc = Environment<>::get().DataConnector();
-
-        auto fieldE = dc.get< FieldE >( FieldE::getName(), true );
-        auto fieldB = dc.get< FieldB >( FieldB::getName(), true );
-
-        EventTask eRfieldE = fieldE->asyncCommunication(__getTransactionEvent());
-        EventTask eRfieldB = fieldB->asyncCommunication(__getTransactionEvent());
-        __setTransactionEvent(eRfieldE);
-        __setTransactionEvent(eRfieldB);
-
-        dc.releaseData( FieldE::getName() );
-        dc.releaseData( FieldB::getName() );
-    }
-
-    static pmacc::traits::StringProperty getStringProperties()
+        /* Directional Splitting conditions:
+         *
+         * using SI units to avoid round off errors
+         *
+         * The compiler is allowed to evaluate an expression those not depends on a template parameter
+         * even if the class is never instantiated. In that case static assert is always
+         * evaluated (e.g. with clang), this results in an error if the condition is false.
+         * http://www.boost.org/doc/libs/1_60_0/doc/html/boost_staticassert.html
+         *
+         * A workaround is to add a template dependency to the expression.
+         * `sizeof(ANY_TYPE) != 0` is always true and defers the evaluation.
+         */
+        PMACC_CASSERT_MSG(DirectionSplitting_Set_dX_equal_dt_times_c____check_your_gridConfig_param_file,
+                          (SI::SPEED_OF_LIGHT_SI * SI::DELTA_T_SI) == SI::CELL_WIDTH_SI &&
+                          (sizeof(T_Dummy) != 0));
+        PMACC_CASSERT_MSG(DirectionSplitting_use_cubic_cells____check_your_gridConfig_param_file,
+                          SI::CELL_HEIGHT_SI == SI::CELL_WIDTH_SI &&
+                          (sizeof(T_Dummy) != 0));
+    #if (SIMDIM == DIM3)
+        PMACC_CASSERT_MSG(DirectionSplitting_use_cubic_cells____check_your_gridConfig_param_file,
+                          SI::CELL_DEPTH_SI == SI::CELL_WIDTH_SI &&
+                          (sizeof(T_Dummy) != 0));
+    #endif
+    };
+} // namespace dirSplitting
+
+    template< typename T_CurrentInterpolation >
+    class DirSplitting: private dirSplitting::ConditionCheck< DirSplitting< T_CurrentInterpolation > >
     {
-        pmacc::traits::StringProperty propList( "name", "DS" );
-        return propList;
-    }
-};
+    private:
+        template<typename OrientationTwist,typename CursorE, typename CursorB, typename GridSize>
+        void propagate(CursorE cursorE, CursorB cursorB, GridSize gridSize) const
+        {
+            using namespace cursor::tools;
+            using namespace pmacc::math;
 
-} // dirSplitting
+            auto gridSizeTwisted = twistComponents<OrientationTwist>(gridSize);
+
+            /* twist components of the supercell */
+            typedef typename CT::TwistComponents<SuperCellSize, OrientationTwist>::type BlockDim;
```

Comment:
you are allowed to remove this `typedef` as well - the file looks like it changed anyway ;)